console-subscriber/src/init.rs

```diff
@@ -0,0 +1,104 @@
+
+type ConsoleSubscriberLayer = Layered<TasksLayer, Layered<EnvFilter, Registry>>;
+
+/// Starts the console subscriber server on its own thread.
+///
+/// This function represents the easiest way to get started using
+/// tokio-console.
+///
+/// **Note**: this function sets the [default `tracing` subscriber][default]
+/// for your application. If you need to add additional layers to this
+/// subscriber, see [`build`].
+///
+/// [default]: https://docs.rs/tracing/latest/tracing/dispatcher/index.html#setting-the-default-subscriber
+///
+/// ## Configuration
+///
+/// Tokio console subscriber is configured with sensible defaults for most
+/// use cases. If you need to tune these parameters, several environmental
+/// configuration variables are available:
+///
+/// | **Environment Variable**            | **Purpose**                                                               | **Default Value** |
+/// |-------------------------------------|---------------------------------------------------------------------------|-------------------|
+/// | `TOKIO_CONSOLE_RETENTION_SECS`      | The number of seconds to accumulate completed tracing data                | 3600s (1h)        |
+/// | `TOKIO_CONSOLE_BIND`                | A HOST:PORT description, such as `localhost:1234`                         | `127.0.0.1:6669`  |
+/// | `TOKIO_CONSOLE_PUBLISH_INTERVAL_MS` | The number of milliseconds to wait between sending updates to the console | 1000ms (1s)       |
+/// | `RUST_LOG`                          | Configure the tracing filter. See [`EnvFilter`] for further information   | `tokio=trace`     |
+///
+/// ## Further customization
+///
+/// To add additional layers or replace the format layer, replace
+/// `console_subscriber::init` with:
+///
+/// ```rust
+/// use tracing_subscriber::prelude::*;
+/// console_subscriber::build()
+///     .with(tracing_subscriber::fmt::layer())
+/// //  .with(..potential additional layer..)
+///     .init();
+/// ```
+
+pub fn init() {
+    build().with(fmt::layer()).init()
+}
+
+/// Builds a new tracing subscriber, but does not yet initialize the
+/// subscriber, allowing for customization by adding additional
+/// layers.
+///
+/// ## Configuration
+///
+/// `console_subscriber::build` supports all of the environmental
+/// configuration described at [`console_subscriber::init`][init]
+///
+/// ## Differences from `init`
+///
+/// **Note**: In order to support customizing the format `build` does
+/// not attach a [`tracing_subscriber::fmt::layer`], unlike [`init`].
+///
+/// Additionally, you must call
+/// [`init`][tracing_subscriber::util::SubscriberInitExt::init] on the
+/// final layer in order to register the subscriber.
+///
+/// ## Examples
+///
+/// ```rust
+/// use tracing_subscriber::prelude::*;
+/// console_subscriber::build()
+///     .with(tracing_subscriber::fmt::layer())
+/// //  .with(...)
+///     .init();
+/// ```
+
+#[must_use = "build() without init() will not attach a tracing subscriber"]
```

Comment:
nit: let's use consistent terminology:
```suggestion
#[must_use = "build() without init() will not set the default tracing subscriber"]
```